components/form/FAutocompleteInput.vue

```diff
@@ -0,0 +1,373 @@
+    :selected-option-text-color="selectedOptionTextColor"
+    :prevent-selection="preventSelection"
+    :disabled="disabled"
+    @select-option="handleSelectOption"
+    :loading="loading"
+  )
+    template(#option="scope")
+      slot.FAutocompleteInput__option(
+        name="option"
+        v-bind="scope"
+      )
+        div(v-html="getSplittedOption(scope.option)")
+    template(#activator="{ openMenu }")
+      FInput(
+        v-model="formattedFieldValue"
+        ref="inputRef"
+        :color="color"
+        :border-color="borderColor"
+        :text-color="textColor"
+        :error-message="errorMessage"
+        :placeholder="placeholder"
+        :placeholder-text-color="placeholderTextColor"
+        :outline-color="outlineColor"
+        :focus-color="focusColor"
+        :focus-border-color="focusBorderColor"
+        :error-color="errorColor"
+        :validate-on-mount="validateOnMount"
+        :validation-trigger="validationTrigger"
+        :disabled="disabled"
+        :clearable="clearable"
+        :rules="[() => isValid]"
+        hide-error-icon
+        hide-hint
+        @focus="e => handleFocus(e, openMenu)"
+        @blur="handleBlur"
+        @change="handleChange"
+        @input="handleInput"
+      )
+</template>
+<style lang="stylus">
+.FAutocompleteInput__option--match
+  padding 0 rem(1)
+  position relative
+
+  &::before
+    content ''
+    position absolute
+    top 0  
+    left 0
+    right 0
+    bottom 0
+    border-radius rem(4)
+    background currentColor
+    opacity .15
+</style>
+
+<script setup lang="ts">
+import FInput from '@/components/form/FInput.vue';
+import FField from '@/components/form/FField.vue';
+import FMenu from '@/components/FMenu.vue';
+
+import { computed, ref, watch } from 'vue';
+import { useFieldWithValidation } from '@/composables/useFieldWithValidation';
+import { useInputEventBindings } from '@/composables/useInputEventBindings';
+import { escapeRegex } from '@/utils/text';
+
+import type { FMenuOption } from '@/components/FMenu.vue';
+
+export interface FAutocompleteInputProps {
+  /**
+   * Input value
+   */
+  modelValue?: string;
+  /**
+   * Propositions based on user input
+   */
+  options?: FMenuOption[];
+  /**
+   * Label, placed on top of select
+   */
+  label?: string;
+  /**
+   * Text color of the label
+   */
+  labelTextColor?: Color;
+  /**
+   * Field name. Used in a form context
+   */
+  name?: string;
+  /**
+   * Validate the number on mount
+   */
+  validateOnMount?: boolean;
+  /**
+   * Text color of the hint
+   */
+  hintTextColor?: Color;
+  /**
+   * Text, hint and caret error color
+   */
+  errorColor?: Color;
+  /**
+   * Hide or not the hint / error message
+   */
+  hideHint?: boolean;
+  /**
+   * A hint to display under the select
+   */
+  hint?: string;
+  /**
+   * Icon, displayed before the hint
+   */
+  hintIcon?: Icon | null;
+  /**
+   * Rules form validation
+   */
+  rules?: ValidationRule | ValidationRule[];
```

Comment:
I see that rules are not plugged to the validation process.
Either you add `props.rules` to the rules given in useFieldWithValidation or in the rules given in FInput, I don't know what should work but my intuition is the second solution, because we could define rules outside (_e.g._ in the case of dynamic option items after an asyn call, the input may have additional rules)

Reply:
They are plugged "implicitly" because we pass props to `useFieldWithValidation` which handles the rules 😄

Reply:
Yes you told me that already ^^